pc/storage/kdtree/kdtree_test.go

```diff
@@ -198,28 +235,352 @@ func TestKDtree(t *testing.T) {
+											dim: 2,
+										},
+									},
+									id:  6,
+									dim: 1,
+								},
+							},
+							id:  0,
+							dim: 0,
+						},
+					},
+				},
+				{
+					pID: 3,
+					expectedTree: &KDTree{
+						Vec3RandomAccessor: it,
+						root: &node{
+							children: [2]*node{
+								&node{
+									children: [2]*node{
+										&node{
+											id:  5,
+											dim: 2,
+										},
+										&node{
+											id:  1,
+											dim: 2,
+										},
+									},
+									id:  4,
+									dim: 1,
+								},
+								&node{
+									children: [2]*node{
+										&node{
+											id:  2,
+											dim: 2,
+										},
+									},
+									id:  6,
+									dim: 1,
+								},
+							},
+							id:  0,
+							dim: 0,
+						},
+					},
+				},
+			},
+		}
+		for name, steps := range testCases {
+			steps := steps
+			t.Run(name, func(t *testing.T) {
+				kdt = New(it)
+				for _, tt := range steps {
+					kdt.deleteNodeImpl(kdt.root, tt.pID, 0)
+					if !kdt.Equal(tt.expectedTree) {
+						t.Fatalf("Expected:\n%v\nGot:\n%v", tt.expectedTree, kdt)
+					}
+				}
+			})
+		}
+	})
+
+	t.Run("DeletePoint", func(t *testing.T) {
+		it := createTestPointCloud(t)
+		testCases := []struct {
+			pID      int
+			hasError bool
+		}{
+			{pID: 6, hasError: false},
+			{pID: -1, hasError: true},
+			{pID: 123, hasError: true},
+		}
+		for _, tt := range testCases {
+			tt := tt
+			t.Run(fmt.Sprintf(
+				"point ID: %d", tt.pID,
+			), func(t *testing.T) {
+				kdt := New(it)
+				err := kdt.DeletePoint(tt.pID)
+				if tt.hasError != (err != nil) {
+					t.Errorf("Expected an error when trying to delete a point that is not in the tree")
+				}
+			})
+		}
+	})
+}
+
+func ExampleKDTree_String() {
+	pp := &pc.PointCloud{
+		PointCloudHeader: pc.PointCloudHeader{
+			Fields: []string{"x", "y", "z"},
+			Size:   []int{4, 4, 4},
+			Type:   []string{"F", "F", "F"},
+			Count:  []int{1, 1, 1},
+			Width:  7,
+			Height: 1,
+		},
+		Points: 7,
+		Data:   make([]byte, 7*4*3),
 	}
+	it, _ := pp.Vec3Iterator()
+	it.SetVec3(mat.Vec3{4, 1, 0}) // 0
+	it.Incr()
+	it.SetVec3(mat.Vec3{2, 2, 1}) // 1
+	it.Incr()
+	it.SetVec3(mat.Vec3{5, 0, 0}) // 2
+	it.Incr()
+	it.SetVec3(mat.Vec3{3, 0, 0}) // 3
+	it.Incr()
+	it.SetVec3(mat.Vec3{0, 1, 0}) // 4
+	it.Incr()
+	it.SetVec3(mat.Vec3{1, 0, 0}) // 5
+	it.Incr()
+	it.SetVec3(mat.Vec3{6, 2, 1}) // 6
+	it, _ = pp.Vec3Iterator()
 	kdt := New(it)
-	ns := &naiveSearch{it}
+	fmt.Println(kdt)
+	// Output:
+	//                     -> (6,2) {6.000, 2.000, 1.000}
+	//           -> (0,1) {4.000, 1.000, 0.000}
+	//                     -> (2,2) {5.000, 0.000, 0.000}
+	// -> (3,0) {3.000, 0.000, 0.000}
+	//                     -> (1,2) {2.000, 2.000, 1.000}
+	//           -> (4,1) {0.000, 1.000, 0.000}
+	//                     -> (5,2) {1.000, 0.000, 0.000}
+}
 
+func testNearestRandomCloud(t *testing.T, it pc.Vec3Iterator, k *KDTree, ns *naiveSearch, nPoints int, width float32) {
+	t.Helper()
```

Comment:
I think this is better not to be helper.
Line number in helper will be hidden and only the caller line will be shown.